src/business_logic/transaction/objects/internal_invoke_function.rs

```diff
@@ -0,0 +1,243 @@
+use felt::Felt;
+
+use crate::{
+    business_logic::{
+        execution::{
+            execution_entry_point::ExecutionEntryPoint,
+            execution_errors::ExecutionError,
+            objects::{CallInfo, TransactionExecutionContext, TransactionExecutionInfo},
+        },
+        fact_state::{
+            in_memory_state_reader::InMemoryStateReader, state::ExecutionResourcesManager,
+        },
+        state::{cached_state::CachedState, update_tracker_state::UpdatesTrackerState},
+        transaction::transaction_errors::TransactionError,
+    },
+    definitions::{
+        constants::EXECUTE_ENTRY_POINT_SELECTOR, general_config::StarknetGeneralConfig,
+        transaction_type::TransactionType,
+    },
+    services::api::contract_class::EntryPointType,
+    utils::{calculate_tx_resources, Address},
+};
+
+pub(crate) struct InternalInvokeFunction {
+    contract_address: Address,
+    entry_point_selector: Felt,
+    _entry_point_type: EntryPointType,
+    calldata: Vec<Felt>,
+    tx_type: TransactionType,
+    version: u64,
+    validate_entry_point_selector: Felt,
+    hash_value: Felt,
+    signature: Vec<Felt>,
+    max_fee: u64,
+    nonce: Felt,
+}
+
+impl InternalInvokeFunction {
+    fn get_execution_context(&self, n_steps: u64) -> TransactionExecutionContext {
+        TransactionExecutionContext::new(
+            self.contract_address.clone(),
+            self.hash_value.clone(),
+            self.signature.clone(),
+            self.max_fee,
+            self.nonce.clone(),
+            n_steps,
+            self.version,
+        )
+    }
+
+    #[allow(dead_code)]
+    fn run_validate_entrypoint(
+        &self,
+        state: &mut CachedState<InMemoryStateReader>,
```

Comment:
We should make this generic @pefontana 
If you think is something we should add after having all the logic implemented, please add an issue in gh and in trello specifying all places where we need to add the generics.

Reply:
Yes, once we modify `ExecutionEntryPoint.execute` so it can receive a generic that implements State it should be trivial to change this

Reply:
#140